src/fparser/two/utils.py

```diff
@@ -554,6 +553,11 @@ def match(startcls, subclasses, endcls, reader,
                 for obj in reversed(content):
                     obj.restore_reader(reader)
                 return
+            if startcls in SYMBOL_TABLES.scoping_unit_classes:
```

Comment:
I think the associated utils tests for BlockBase should check that SYMBOLS_TABLES is set and not set for different given examples. I appreciate this will be covered by tests in different files but I prefer to have a test-file cover all code in the file it is testing.

Reply:
Done.